src/components/AuthNavigationHandler.tsx

```diff
@@ -0,0 +1,37 @@
+import { authCommands } from 'lib/commandBarConfigs/authCommandConfig'
+import { PATHS } from 'lib/paths'
+import { useAuthState } from 'machines/appMachine'
+import { commandBarActor, useCommandBarState } from 'machines/commandBarMachine'
+import { useEffect } from 'react'
+import { useLocation, useNavigate } from 'react-router-dom'
+
+/**
+ * A simple HOC that listens to the auth state of the app and navigates
+ * accordingly.
+ */
+export function AuthNavigationHandler({
```

Comment:
No way we could make this a hook eh? I see how you're using it but I believe it could be used in the `Router` component. It feels weird to use this as a HOC when it's not modifying the component(s) at all.

Reply:
You're absolutely right, I should have made this a hook from the jump. I've just made that change.